spoti-friends/src/spotify/spotifyAPI/SpotifyAPI.swift

```diff
@@ -51,10 +55,17 @@ extension SpotifyAPI {
     /// Creates and returns the URLRequest object to corresponding Spotify API `endpoint`
     ///
     /// `endpoint` should be prepended with a "/".
-    private func createRequestTo(endpoint: String, accessToken: String, method: String) throws -> URLRequest {
-        guard let url = URL(string: APIConstants.host + endpoint) else { throw URLError(.badURL)}
+    private func createRequestTo(endpoint: String, accessToken: String, method: RequestMethod, queryParams: [URLQueryItem]) throws -> URLRequest {
```

Comment:
change `endpoint` type to `APIEndpoint` and then use `.rawValue` here. Then when passing it into this function, just pass in the `APIEndpoint` value, not the raw `String` value.